Build Failure Due to CMake Version Requirement (>= 3.5)

When I try to compile this library, the following error occurs:

```
  CMake Error at CMakeLists.txt:15 (cmake_minimum_required):
    Compatibility with CMake < 3.5 has been removed from CMake.

    Update the VERSION argument <min> value.  Or, use the <min>...<max> syntax
    to tell CMake that the project requires at least <min> but has been updated
    to work with policies introduced by <max> or earlier.

    Or, add -DCMAKE_POLICY_VERSION_MINIMUM=3.5 to try configuring anyway.
```

<details>
<summary>Here is the complete log</summary>

```
$ cargo build
    Updating crates.io index
     Locking 21 packages to latest compatible versions
      Adding num-derive v0.3.3 (available: v0.4.2)
      Adding prost v0.11.9 (available: v0.13.5)
      Adding prost-derive v0.11.9 (available: v0.13.5)
      Adding thiserror v1.0.69 (available: v2.0.12)
  Downloaded either v1.15.0
  Downloaded pkg-config v0.3.32
  Downloaded anyhow v1.0.98
  Downloaded unicode-ident v1.0.18
  Downloaded cc v1.2.19
  Downloaded libc v0.2.172
  Downloaded 6 crates (1040.1 KB) in 0.53s
   Compiling proc-macro2 v1.0.94
   Compiling unicode-ident v1.0.18
   Compiling syn v1.0.109
   Compiling shlex v1.3.0
   Compiling anyhow v1.0.98
   Compiling pkg-config v0.3.32
   Compiling either v1.15.0
   Compiling autocfg v1.4.0
   Compiling cc v1.2.19
   Compiling itertools v0.10.5
   Compiling libc v0.2.172
   Compiling num-traits v0.2.19
   Compiling thiserror v1.0.69
   Compiling cmake v0.1.54
   Compiling bytes v1.10.1
   Compiling sentencepiece-sys v0.11.2 (/Volumes/ExtremeSSD/Workspace/src/github.com/danieldk/sentencepiece/sentencepiece-sys)
   Compiling quote v1.0.40
   Compiling syn v2.0.100
error: failed to run custom build command for `sentencepiece-sys v0.11.2 (/Volumes/ExtremeSSD/Workspace/src/github.com/danieldk/sentencepiece/sentencepiece-sys)`

Caused by:
  process didn't exit successfully: `/Volumes/ExtremeSSD/Workspace/src/github.com/danieldk/sentencepiece/target/debug/build/sentencepiece-sys-f19ff60f623aa46a/build-script-build` (exit status: 101)
  --- stdout
  cargo:rerun-if-env-changed=SENTENCEPIECE_NO_PKG_CONFIG
  cargo:rerun-if-env-changed=PKG_CONFIG_aarch64-apple-darwin
  cargo:rerun-if-env-changed=PKG_CONFIG_aarch64_apple_darwin
  cargo:rerun-if-env-changed=HOST_PKG_CONFIG
  cargo:rerun-if-env-changed=PKG_CONFIG
  cargo:rerun-if-env-changed=SENTENCEPIECE_STATIC
  cargo:rerun-if-env-changed=SENTENCEPIECE_DYNAMIC
  cargo:rerun-if-env-changed=PKG_CONFIG_ALL_STATIC
  cargo:rerun-if-env-changed=PKG_CONFIG_ALL_DYNAMIC
  cargo:rerun-if-env-changed=PKG_CONFIG_PATH_aarch64-apple-darwin
  cargo:rerun-if-env-changed=PKG_CONFIG_PATH_aarch64_apple_darwin
  cargo:rerun-if-env-changed=HOST_PKG_CONFIG_PATH
  cargo:rerun-if-env-changed=PKG_CONFIG_PATH
  cargo:rerun-if-env-changed=PKG_CONFIG_LIBDIR_aarch64-apple-darwin
  cargo:rerun-if-env-changed=PKG_CONFIG_LIBDIR_aarch64_apple_darwin
  cargo:rerun-if-env-changed=HOST_PKG_CONFIG_LIBDIR
  cargo:rerun-if-env-changed=PKG_CONFIG_LIBDIR
  cargo:rerun-if-env-changed=PKG_CONFIG_SYSROOT_DIR_aarch64-apple-darwin
  cargo:rerun-if-env-changed=PKG_CONFIG_SYSROOT_DIR_aarch64_apple_darwin
  cargo:rerun-if-env-changed=HOST_PKG_CONFIG_SYSROOT_DIR
  cargo:rerun-if-env-changed=PKG_CONFIG_SYSROOT_DIR
  cargo:rerun-if-env-changed=SENTENCEPIECE_STATIC
  cargo:rerun-if-env-changed=SENTENCEPIECE_DYNAMIC
  cargo:rerun-if-env-changed=PKG_CONFIG_ALL_STATIC
  cargo:rerun-if-env-changed=PKG_CONFIG_ALL_DYNAMIC
  cargo:rerun-if-env-changed=PKG_CONFIG_PATH_aarch64-apple-darwin
  cargo:rerun-if-env-changed=PKG_CONFIG_PATH_aarch64_apple_darwin
  cargo:rerun-if-env-changed=HOST_PKG_CONFIG_PATH
  cargo:rerun-if-env-changed=PKG_CONFIG_PATH
  cargo:rerun-if-env-changed=PKG_CONFIG_LIBDIR_aarch64-apple-darwin
  cargo:rerun-if-env-changed=PKG_CONFIG_LIBDIR_aarch64_apple_darwin
  cargo:rerun-if-env-changed=HOST_PKG_CONFIG_LIBDIR
  cargo:rerun-if-env-changed=PKG_CONFIG_LIBDIR
  cargo:rerun-if-env-changed=PKG_CONFIG_SYSROOT_DIR_aarch64-apple-darwin
  cargo:rerun-if-env-changed=PKG_CONFIG_SYSROOT_DIR_aarch64_apple_darwin
  cargo:rerun-if-env-changed=HOST_PKG_CONFIG_SYSROOT_DIR
  cargo:rerun-if-env-changed=PKG_CONFIG_SYSROOT_DIR
  OPT_LEVEL = Some(0)
  OUT_DIR = Some(/Volumes/ExtremeSSD/Workspace/src/github.com/danieldk/sentencepiece/target/debug/build/sentencepiece-sys-5d45db0dc393c270/out)
  TARGET = Some(aarch64-apple-darwin)
  HOST = Some(aarch64-apple-darwin)
  cargo:rerun-if-env-changed=CC_aarch64-apple-darwin
  CC_aarch64-apple-darwin = None
  cargo:rerun-if-env-changed=CC_aarch64_apple_darwin
  CC_aarch64_apple_darwin = None
  cargo:rerun-if-env-changed=HOST_CC
  HOST_CC = None
  cargo:rerun-if-env-changed=CC
  CC = None
  cargo:rerun-if-env-changed=CC_ENABLE_DEBUG_OUTPUT
  RUSTC_WRAPPER = None
  cargo:rerun-if-env-changed=CRATE_CC_NO_DEFAULTS
  CRATE_CC_NO_DEFAULTS = None
  DEBUG = Some(true)
  cargo:rerun-if-env-changed=MACOSX_DEPLOYMENT_TARGET
  MACOSX_DEPLOYMENT_TARGET = None
  cargo:rerun-if-env-changed=CFLAGS
  CFLAGS = None
  cargo:rerun-if-env-changed=HOST_CFLAGS
  HOST_CFLAGS = None
  cargo:rerun-if-env-changed=CFLAGS_aarch64_apple_darwin
  CFLAGS_aarch64_apple_darwin = None
  cargo:rerun-if-env-changed=CFLAGS_aarch64-apple-darwin
  CFLAGS_aarch64-apple-darwin = None
  CARGO_ENCODED_RUSTFLAGS = Some()
  CMAKE_TOOLCHAIN_FILE_aarch64-apple-darwin = None
  CMAKE_TOOLCHAIN_FILE_aarch64_apple_darwin = None
  HOST_CMAKE_TOOLCHAIN_FILE = None
  CMAKE_TOOLCHAIN_FILE = None
  CMAKE_GENERATOR_aarch64-apple-darwin = None
  CMAKE_GENERATOR_aarch64_apple_darwin = None
  HOST_CMAKE_GENERATOR = None
  CMAKE_GENERATOR = None
  CMAKE_PREFIX_PATH_aarch64-apple-darwin = None
  CMAKE_PREFIX_PATH_aarch64_apple_darwin = None
  HOST_CMAKE_PREFIX_PATH = None
  CMAKE_PREFIX_PATH = None
  CMAKE_aarch64-apple-darwin = None
  CMAKE_aarch64_apple_darwin = None
  HOST_CMAKE = None
  CMAKE = None
  running: cd "/Volumes/ExtremeSSD/Workspace/src/github.com/danieldk/sentencepiece/target/debug/build/sentencepiece-sys-5d45db0dc393c270/out/build" && CMAKE_PREFIX_PATH="" LC_ALL="C" "cmake" "/Volumes/ExtremeSSD/Workspace/src/github.com/danieldk/sentencepiece/sentencepiece-sys/source" "-DCMAKE_OSX_ARCHITECTURES=arm64" "-DCMAKE_INSTALL_PREFIX=/Volumes/ExtremeSSD/Workspace/src/github.com/danieldk/sentencepiece/target/debug/build/sentencepiece-sys-5d45db0dc393c270/out" "-DCMAKE_C_FLAGS= -ffunction-sections -fdata-sections -fPIC --target=arm64-apple-macosx -mmacosx-version-min=15.4" "-DCMAKE_C_COMPILER=/usr/bin/cc" "-DCMAKE_CXX_FLAGS= -ffunction-sections -fdata-sections -fPIC --target=arm64-apple-macosx -mmacosx-version-min=15.4" "-DCMAKE_CXX_COMPILER=/usr/bin/c++" "-DCMAKE_ASM_FLAGS= -ffunction-sections -fdata-sections -fPIC --target=arm64-apple-macosx -mmacosx-version-min=15.4" "-DCMAKE_ASM_COMPILER=/usr/bin/cc" "-DCMAKE_BUILD_TYPE=Debug"
  -- Configuring incomplete, errors occurred!

  --- stderr
  CMake Error at CMakeLists.txt:15 (cmake_minimum_required):
    Compatibility with CMake < 3.5 has been removed from CMake.

    Update the VERSION argument <min> value.  Or, use the <min>...<max> syntax
    to tell CMake that the project requires at least <min> but has been updated
    to work with policies introduced by <max> or earlier.

    Or, add -DCMAKE_POLICY_VERSION_MINIMUM=3.5 to try configuring anyway.



  thread 'main' panicked at /Users/junpei/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/cmake-0.1.54/src/lib.rs:1119:5:

  command did not execute successfully, got: exit status: 1

  build script failed, must exit now
  note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
warning: build failed, waiting for other jobs to finish...
```
</details>

### Environment
- **OS**: macOS 15.4
- **Cargo version**: `cargo 1.86.0 (adf9b6ad1 2025-02-28)`
- **Rustc version**: `rustc 1.86.0 (05f9846f8 2025-03-31)`
- **CMake version**: 4.0.1

--- a/sentencepiece-sys/build.rs
+++ b/sentencepiece-sys/build.rs
@@ -7,6 +7,7 @@ macro_rules! feature(($name:expr) => (env::var(concat!("CARGO_FEATURE_", $name))
 
 fn build_sentencepiece(builder: &mut Build) {
     let mut config = Config::new("source");
+    config.define("CMAKE_POLICY_VERSION_MINIMUM", "3.5");
     if builder.get_compiler().is_like_msvc() {
         config.profile("Release");
         if env::var("CARGO_CFG_TARGET_FEATURE")

